config/config-api/src/com/thoughtworks/go/config/exceptions/GoConfigInvalidException.java

```diff
@@ -12,23 +12,34 @@
  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  * See the License for the specific language governing permissions and
  * limitations under the License.
- *************************GO-LICENSE-END***********************************/
+ */
 
 package com.thoughtworks.go.config.exceptions;
 
-import java.util.List;
-
 import com.thoughtworks.go.config.CruiseConfig;
 import com.thoughtworks.go.domain.ConfigErrors;
+import com.thoughtworks.go.util.ListUtil;
+
+import java.util.List;
 
 public class GoConfigInvalidException extends RuntimeException {
     private final CruiseConfig cruiseConfig;
 
-    public GoConfigInvalidException(CruiseConfig cruiseConfig, List<ConfigErrors> allErrors) {
-        super(allErrors.get(0).firstError());
+    public GoConfigInvalidException(CruiseConfig cruiseConfig, String error) {
+        super(error);
         this.cruiseConfig = cruiseConfig;
     }
 
+    public GoConfigInvalidException(CruiseConfig cruiseConfig, List<ConfigErrors> allErrors) {
+        this(cruiseConfig, ListUtil.join(ListUtil.map(allErrors, new ListUtil.Transformer<String>() {
+            @Override
+            public String transform(Object obj) {
+                ConfigErrors errors = (ConfigErrors) obj;
+                return ListUtil.join(errors.getAll());
+            }
+        })));
+    }
+
```

Comment:
Looking for usages of this class and constructor. I feel we need a better API
for e.g. to construct this exception one has to do this 
```
    private GoConfigInvalidException getGoConfigInvalidException() {
        ConfigErrors configErrors = new ConfigErrors();
        configErrors.add("command", "command cannot be empty");
        ArrayList<ConfigErrors> list = new ArrayList<ConfigErrors>();
        list.add(configErrors);
        return new GoConfigInvalidException(new BasicCruiseConfig(), list);
    }

```
(this is a test)

But other usages goes like
```
throw new GoConfigInvalidException(preprocessedConfig, new ArrayList<ConfigErrors>());
```

Also ideally we should have just wrapped/extended the ```List<ConfigErrors>``` onto something like ```AllConfigErrors``` and wrapped this method onto that

```
AllConfigErrors.errorAsString
 {Fancy loops}
```

Atleast

```
              ConfigErrors errors = (ConfigErrors) obj;
               return ListUtil.join(errors.getAll());
```
Could have been

```
              ConfigErrors errors = (ConfigErrors) obj;
               return errors.asString();
```